src/spawn.ts

```diff
@@ -147,7 +147,7 @@ interface Spawning {
      * An array with the spawn directions
      * @see http://docs.screeps.com/api/#StructureSpawn.Spawning.setDirections
      */
-    directions: DirectionConstant[];
+    directions: DirectionConstant[] | undefined;
 
```

Comment:
```suggestion
    directions?: DirectionConstant[];
```
use `?:` instead of `| undefined` to express optional property